### PR TITLE
[MB-3015] Fix duplicate nesting of Change Role UI

### PR DESCRIPTION
**Pull request by suzubara**
## Description
Fixes MB-3015 by extracting the "Change user role" link from `<PrivateRoute>` and putting it in the Office index instead (which currently serves to render the layout for all Office pages). Other changes are:
- moved `PrivateRoute` into the new component structure
- removed the `hideSwitcher` prop which had been added as a temporary workaround
- modified `PrivateRoute` so that if the user is not logged in, it actually redirects them to a `/sign-in` path (instead of rendering the `SignIn` component at the attempted path)

#### User with multiple roles
1. Log into the Office app as a user with multiple roles
1. Verify you can still click on "Change user role" in the top left to change which role you're acting as
1. Verify the "Change user role" link appears _above_ the office.move.mil header
1. Verify when you navigate to one of the TOO Move Details pages, there is only one instance of the "Change user role" link

#### User with one role
1. Log into the Office app as a user with only one role, and verify you do not see the "Change user role" link at all.

#### Not logged in
1. Try to access an Office app URL without being logged in, and verify you are redirected to the Sign In page.

## Code Review Verification Steps

* [ ] There are no aXe warnings for UI.
* [ ] This works in [Supported Browsers and their phone views](https://github.com/transcom/mymove/tree/master/docs/adr/0016-Browser-Support.md) (Chrome, Firefox, IE, Edge).
* [ ] User facing changes have been reviewed by design.
* [ ] Request review from a member of a different team.
* [ ] Have the Jira acceptance criteria been met for this change?

## References

* [Jira story](https://dp3.atlassian.net/browse/MB-3015) for this change

## Screenshots

image

### CODE REVIEW

File: src/pages/Office/index.test.jsx [hunk 1]
```diff
@@ -43,11 +43,55 @@ describe('OfficeWrapper', () => {
...
+          expect(app.containsMatchingElement(<a href="/select-application">Change user role</a>)).toEqual(false);
+        });
+      });
+    });
   });
 
   describe('routing', () => {
     // TODO - expects should look for actual component content instead of the route path
     // Might have to add testing-library for this because something about enzyme + Suspense + routes are not rendering content
+    // I FIGURED OUT HOW - need to mock the getCurrentUserInfo (this sets loading back to true and prevents content from rendering)
```
Comment (duncan-truss) [inline src/pages/Office/index.test.jsx:94]: 👏 🎉

File: src/pages/Office/index.jsx [hunk 2]
```diff
@@ -174,6 +188,11 @@ OfficeWrapper.propTypes = {
   }),
   location: LocationShape,
   userIsLoggedIn: PropTypes.bool,
+  userRoles: PropTypes.arrayOf(
```
Comment (duncan-truss) [inline src/pages/Office/index.jsx:191]: I think you made the UserRolesShape to replace this

**Comment by duncan-truss** [comment]
I'm getting an error when logging to the office as a PPM user. Are you seeing the same thing?

Screen Shot 2020-06-30 at 5 50 32 PM

**Comment by suzubara** [comment]
@duncan-truss Hm I am not seeing that.. how fresh is your database? I know there have been some issues popping up where stale users don't have roles defined..

**Comment by duncan-truss** [comment]
> @duncan-truss Hm I am not seeing that.. how fresh is your database? I know there have been some issues popping up where stale users don't have roles defined..

I reset my database, reran migrations, and loaded the e2e basic scenario and got the same error.  It can be worked around if I go in the admin app and assign that user the ppm role. You can see where it tries to create the role here https://github.com/transcom/mymove/blob/master/pkg/testdatagen/scenario/e2ebasic.go#L41

Screen Shot 2020-07-01 at 4 49 47 PM

**Comment by duncan-truss** [comment]
@suzubara Oh maybe I haven't rebuilt that binary and I needed to make build_tools again 🤦

**Pull request merged**